docs/mllib-feature-extraction.md

```diff
@@ -44,6 +44,8 @@ To reduce the chance of collision, we can increase the target feature dimension,
 the number of buckets of the hash table.
 The default feature dimension is `$2^{20} = 1,048,576$`.
 
+We recommend users to adapt to the DataFrame-based API in [ML user guide on TF-IDF](ml-features.html#tf-idf).
```

Comment:
How strongly do we want to push users to the DF API? We could start putting this as a `**Note**` just below the section heading (e.g. right below `## TF-IDF` above), which would make it a bit more "aggressive", but we are explicitly making the RDD-based APIs maintenance only.

cc @jkbradley